--- include/vec128dp.h
+++ include/vec128dp.h
@@ -466,7 +466,7 @@
 
 
 /* Floor 2      64-bit doubles */
-__m128d vec_Floor2dp (__m128d v){
+VECLIB_INLINE __m128d vec_Floor2dp (__m128d v){
   return vec_floor(v);
 }
 
--- include/vec128int.h
+++ include/vec128int.h
@@ -1046,11 +1046,11 @@
 }
 
 /* Max  4 32-bit signed ints */
-__m128i vec_Max4sw (__m128i left, __m128i right){
+VECLIB_INLINE __m128i vec_Max4sw (__m128i left, __m128i right){
   return (__m128i)vec_max((vector signed int)left, (vector signed int)right);
 }
 /* Min  4 32-bit signed ints */
-__m128i vec_Min4sw (__m128i left, __m128i right){
+VECLIB_INLINE __m128i vec_Min4sw (__m128i left, __m128i right){
   return (__m128i)vec_min((vector signed int)left, (vector signed int)right);
 }
 
--- include/vec128sp.h
+++ include/vec128sp.h
@@ -1031,7 +1031,7 @@
 }
 
 /* Floor 4      32-bit floats */
-__m128  vec_Floor4sp (__m128  v){
+VECLIB_INLINE __m128  vec_Floor4sp (__m128  v){
   return vec_floor(v);
 }
 
--- include/vecmisc.h
+++ include/vecmisc.h
@@ -175,7 +175,7 @@
 }
 
 /* Zero upper half of all 8 or 16 YMMM registers */
-void vec_zeroallupper (void)  
+VECLIB_INLINE void vec_zeroallupper (void)  
 {
   /* no-op */
 }
