# Ensures "bedmap --version" returns with exit code 0, as expected
--- applications/bed/bedmap/src/Bedmap.cpp
+++ applications/bed/bedmap/src/Bedmap.cpp
@@ -168,6 +168,7 @@
     std::cout << "  citation: " << BedMap::citation << std::endl;
     std::cout << "  version:  " << BedMap::version << std::endl;
     std::cout << "  authors:  " << BedMap::authors << std::endl;
+    return(EXIT_SUCCESS);
   } catch(const BedMap::NoInput& ni) { // show usage and exit failure
     std::cerr << BedMap::prognm << std::endl;
     std::cerr << "  citation: " << BedMap::citation << std::endl;
