--- src/configure.in
+++ src/configure.in
@@ -124,7 +124,7 @@
    
     if test "x$enable_threads" == "xyes"
     then
-        AC_CHECK_LIB(pthread,pthread_create,[THREAD_LIB="-lpthread"],
+        AC_CHECK_LIB(pthread,pthread_create,[LDFLAGS=${LDFLAGS}" -lpthread"],
         [AC_MSG_ERROR([pthreads library not found])])
         AC_DEFINE(THREADS_ENABLED)
     fi
@@ -142,9 +142,9 @@
 AC_LANG_C
 
 dnl check readline
-AC_CHECK_LIB([readline], [readline])
+AC_CHECK_LIB([readline], [readline], , ,[-lncurses])
 if test $ac_cv_lib_readline_readline = yes; then
-    LDFLAGS=${LDFLAGS}" -lreadline"
+    LDFLAGS=${LDFLAGS}" -lreadline -lncurses"
     AC_MSG_CHECKING(for rl_completion_matches function)
     AC_LINK_IFELSE(
     [AC_LANG_SOURCE([
