--- Makefile
+++ Makefile
@@ -1,7 +1,7 @@
 CC=			gcc
-CFLAGS=		-g -Wall -O2
+CFLAGS+=		-Wall
 #LDFLAGS=		-Wl,-rpath,\$$ORIGIN/../lib
-DFLAGS=		-D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_USE_KNETFILE -D_CURSES_LIB=1
+DFLAGS=		-D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_USE_KNETFILE -D_CURSES_LIB=0
 KNETFILE_O=	knetfile.o
 LOBJS=		bgzf.o kstring.o bam_aux.o bam.o bam_import.o sam.o bam_index.o	\
 			bam_pileup.o bam_lpileup.o bam_md.o razf.o faidx.o bedidx.o \
@@ -15,7 +15,7 @@ PROG=		samtools
 INCLUDES=	-I.
 SUBDIRS=	. bcftools misc
 LIBPATH=
-LIBCURSES=	-lcurses # -lXCurses
+PREFIX ?=	/usr/local
 
 .SUFFIXES:.c .o
 .PHONY: all lib
@@ -28,7 +28,7 @@ all-recur lib-recur clean-recur cleanlocal-recur install-recur:
 		wdir=`pwd`; \
 		list='$(SUBDIRS)'; for subdir in $$list; do \
 			cd $$subdir; \
-			$(MAKE) CC="$(CC)" DFLAGS="$(DFLAGS)" CFLAGS="$(CFLAGS)" \
+			$(MAKE) PREFIX="$(PREFIX)" CC="$(CC)" DFLAGS="$(DFLAGS)" CFLAGS="$(CFLAGS)" \
 				INCLUDES="$(INCLUDES)" LIBPATH="$(LIBPATH)" $$target || exit 1; \
 			cd $$wdir; \
 		done;
@@ -37,9 +37,16 @@ all:$(PROG)
 
 .PHONY:all lib clean cleanlocal
 .PHONY:all-recur lib-recur clean-recur cleanlocal-recur install-recur
+.PHONY:install
 
 lib:libbam.a
 
+install: all
+	install -d "$(PREFIX)/bin" "$(PREFIX)/lib" "$(PREFIX)/include/samtools"
+	install $(PROG) "$(PREFIX)/bin"
+	install libbam.a "$(PREFIX)/lib"
+	install bam.h sam.h bgzf.h "$(PREFIX)/include/samtools"
+
 libbam.a:$(LOBJS)
 		$(AR) -csru $@ $(LOBJS)
 
@@ -47,10 +54,10 @@ samtools:lib-recur $(AOBJS)
 		$(CC) $(CFLAGS) -o $@ $(AOBJS) $(LDFLAGS) libbam.a -Lbcftools -lbcf $(LIBPATH) $(LIBCURSES) -lm -lz -lpthread
 
 razip:razip.o razf.o $(KNETFILE_O)
-		$(CC) $(CFLAGS) -o $@ $^ -lz
+		$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) -lz
 
 bgzip:bgzip.o bgzf.o $(KNETFILE_O)
-		$(CC) $(CFLAGS) -o $@ $^ -lz -lpthread
+		$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) -lz -lpthread
 
 bgzf.o:bgzf.c bgzf.h
 		$(CC) -c $(CFLAGS) $(DFLAGS) -DBGZF_CACHE $(INCLUDES) bgzf.c -o $@
@@ -81,10 +88,10 @@ faidx_main.o:faidx.h razf.h
 
 
 libbam.1.dylib-local:$(LOBJS)
-		libtool -dynamic $(LOBJS) -o libbam.1.dylib -lc -lz
+		libtool -dynamic $(LOBJS) -o libbam.1.dylib $(LDFLAGS) -lc -lz
 
 libbam.so.1-local:$(LOBJS)
-		$(CC) -shared -Wl,-soname,libbam.so -o libbam.so.1 $(LOBJS) -lc -lz
+		$(CC) -shared -Wl,-soname,libbam.so -o libbam.so.1 $(LOBJS) $(LDFLAGS) -lc -lz
 
 dylib:
 		@$(MAKE) cleanlocal; \
--- bcftools/Makefile
+++ bcftools/Makefile
@@ -7,8 +7,10 @@ AOBJS=		call1.o main.o $(OMISC)/kstring.o $(OMISC)/bgzf.o $(OMISC)/knetfile.o $(
 PROG=		bcftools
 INCLUDES=	
 SUBDIRS=	.
+PREFIX ?= /usr/local
 
 .SUFFIXES:.c .o
+.PHONY:install
 
 .c.o:
 		$(CC) -c $(CFLAGS) $(DFLAGS) -I.. $(INCLUDES) $< -o $@
@@ -25,13 +27,17 @@ all-recur lib-recur clean-recur cleanlocal-recur install-recur:
 
 all:$(PROG)
 
+install: all
+	install -d $(PREFIX)/bin
+	install $(PROG) vcfutils.pl $(PREFIX)/bin
+
 lib:libbcf.a
 
 libbcf.a:$(LOBJS)
 		$(AR) -csru $@ $(LOBJS)
 
 bcftools:lib $(AOBJS)
-		$(CC) $(CFLAGS) -o $@ $(AOBJS) -L. $(LIBPATH) -lbcf -lm -lz -lpthread
+		$(CC) $(CFLAGS) -o $@ $(AOBJS) $(LDFLAGS) -L. $(LIBPATH) -lbcf -lm -lz -lpthread
 
 bcf.o:bcf.h
 vcf.o:bcf.h
--- examples/Makefile
+++ examples/Makefile
@@ -40,7 +40,7 @@ ex1.bcf:ex1.bam ex1.fa.fai
 		(cd ..; make libbam.a)
 
 calDepth:../libbam.a calDepth.c
-		gcc -g -Wall -O2 -I.. calDepth.c -o $@ -L.. -lbam -lm -lz
+		$(CC) $(CFLAGS) -I.. calDepth.c -o $@ $(LDFLAGS) -L.. -lbam -lm -lz
 
 clean:
 		rm -fr *.bam *.bai *.glf* *.fai *.pileup* *~ calDepth *.dSYM ex1*.rg ex1.bcf
--- misc/Makefile
+++ misc/Makefile
@@ -7,14 +7,20 @@ OBJS=
 PROG=		md5sum-lite md5fa maq2sam-short maq2sam-long ace2sam wgsim bamcheck
 INCLUDES=	-I..
 SUBDIRS=	.
+PREFIX ?= /usr/local
 
 .SUFFIXES:.c .o
+.PHONY: all install
 
 .c.o:
 		$(CC) -c $(CFLAGS) $(DFLAGS) $(INCLUDES) $< -o $@
 
 all:$(PROG)
 
+install: all
+	install -d $(PREFIX)/bin
+	install $(PROG) varfilter.py plot-bamcheck *.pl $(PREFIX)/bin
+
 lib-recur all-recur clean-recur cleanlocal-recur install-recur:
 		@target=`echo $@ | sed s/-recur//`; \
 		wdir=`pwd`; \
@@ -28,19 +34,19 @@ lib-recur all-recur clean-recur cleanlocal-recur install-recur:
 lib:
 
 bamcheck:bamcheck.o
-		$(CC) $(CFLAGS) -o $@ bamcheck.o -L.. -lm -lbam -lpthread -lz
+		$(CC) $(CFLAGS) -o $@ bamcheck.o $(LDFLAGS) -L.. -lm -lbam -lpthread -lz
 
 bamcheck.o:bamcheck.c ../faidx.h ../khash.h ../sam.h ../razf.h
 		$(CC) $(CFLAGS) -c -I.. -o $@ bamcheck.c
 
 ace2sam:ace2sam.o
-		$(CC) $(CFLAGS) -o $@ ace2sam.o -lz
+		$(CC) $(CFLAGS) -o $@ ace2sam.o $(LDFLAGS) -lz
 
 wgsim:wgsim.o
-		$(CC) $(CFLAGS) -o $@ wgsim.o -lm -lz
+		$(CC) $(CFLAGS) -o $@ wgsim.o $(LDFLAGS) -lm -lz
 
 md5fa:md5.o md5fa.o md5.h ../kseq.h
-		$(CC) $(CFLAGS) -o $@ md5.o md5fa.o -lz
+		$(CC) $(CFLAGS) -o $@ md5.o md5fa.o $(LDFLAGS) -lz
 
 md5sum-lite:md5sum-lite.o
 		$(CC) $(CFLAGS) -o $@ md5sum-lite.o
@@ -49,10 +55,10 @@ md5sum-lite.o:md5.c md5.h
 		$(CC) -c $(CFLAGS) -DMD5SUM_MAIN -o $@ md5.c
 
 maq2sam-short:maq2sam.c
-		$(CC) $(CFLAGS) -o $@ maq2sam.c -lz
+		$(CC) $(CFLAGS) -o $@ maq2sam.c $(LDFLAGS) -lz
 
 maq2sam-long:maq2sam.c
-		$(CC) $(CFLAGS) -DMAQ_LONGREADS -o $@ maq2sam.c -lz
+		$(CC) $(CFLAGS) -DMAQ_LONGREADS -o $@ maq2sam.c $(LDFLAGS) -lz
 
 md5fa.o:md5.h md5fa.c
 		$(CC) $(CFLAGS) -c -I.. -o $@ md5fa.c
