--- convert-sam-for-rsem
+++ convert-sam-for-rsem
@@ -6,10 +6,7 @@
 
 use FindBin;
 use lib $FindBin::RealBin;
-use rsem_perl_utils qw(runCommand getSAMTOOLS);
-
-use Env qw(@PATH);
-@PATH = ($FindBin::RealBin, "$FindBin::RealBin/" . getSAMTOOLS(), @PATH);
+use rsem_perl_utils qw(runCommand);
 
 use strict;
 
--- rsem-calculate-expression
+++ rsem-calculate-expression
@@ -4,12 +4,8 @@
 use Pod::Usage;
 use File::Basename;
 use FindBin;
-use lib $FindBin::RealBin;
-use rsem_perl_utils qw(runCommand collectResults showVersionInfo getSAMTOOLS);
-
-use Env qw(@PATH);
-
-@PATH = ($FindBin::RealBin, "$FindBin::RealBin/" . getSAMTOOLS(), @PATH);
+use lib '@@share_dir@@';
+use rsem_perl_utils qw(runCommand collectResults showVersionInfo);
 
 use strict;
 
--- rsem-generate-ngvector
+++ rsem-generate-ngvector
@@ -4,7 +4,7 @@
 use Pod::Usage;
 
 use FindBin;
-use lib $FindBin::RealBin;
+use lib '@@share_dir@@';
 use rsem_perl_utils;
 
 use Env qw(@PATH);
--- rsem-plot-transcript-wiggles
+++ rsem-plot-transcript-wiggles
@@ -4,11 +4,8 @@
 use Pod::Usage;
 
 use FindBin;
-use lib $FindBin::RealBin;
-use rsem_perl_utils qw(runCommand getSAMTOOLS);
-
-use Env qw(@PATH);
-@PATH = ($FindBin::RealBin, "$FindBin::RealBin/" . getSAMTOOLS(), @PATH);
+use lib '@@share_dir@@';
+use rsem_perl_utils qw(runCommand);
 
 use strict;
 
--- rsem-prepare-reference
+++ rsem-prepare-reference
@@ -4,7 +4,7 @@
 use Pod::Usage;	
 use File::Basename;
 use FindBin;
-use lib $FindBin::RealBin;
+use lib '@@share_dir@@';
 use rsem_perl_utils;
 
 use Env qw(@PATH);
--- rsem-run-ebseq
+++ rsem-run-ebseq
@@ -4,7 +4,7 @@
 use Pod::Usage;
 
 use FindBin;
-use lib $FindBin::RealBin;
+use lib '@@share_dir@@';
 use rsem_perl_utils;
 
 use Env qw(@PATH);
--- rsem_perl_utils.pm
+++ rsem_perl_utils.pm
@@ -10,7 +10,7 @@
 our @EXPORT_OK = qw(runCommand collectResults showVersionInfo getSAMTOOLS);
 
 my $version = "RSEM v1.2.29"; # Update version info here
-my $samtools = "samtools-1.3"; # If update to another version of SAMtools, need to change this
+my $samtools = "samtools"; # If update to another version of SAMtools, need to change this
 
 # command, {err_msg}
 sub runCommand {
