--- Makefile
+++ Makefile
@@ -2,11 +2,19 @@
 CFLAGS  += -Wall -O3 -ffast-math -I.
 LDFLAGS +=
 PROGRAMS = rsem-extract-reference-transcripts rsem-synthesis-reference-transcripts rsem-preref rsem-parse-alignments rsem-build-read-index rsem-run-em rsem-tbam2gbam rsem-run-gibbs rsem-calculate-credibility-intervals rsem-simulate-reads rsem-bam2wig rsem-get-unique rsem-bam2readdepth rsem-sam-validator rsem-scan-for-paired-end-reads
+SCRIPTS = convert-sam-for-rsem extract-transcript-to-gene-map-from-trinity rsem-calculate-expression rsem-control-fdr rsem-generate-data-matrix rsem-generate-ngvector rsem-gen-transcript-plots rsem_perl_utils.pm rsem-plot-model rsem-plot-transcript-wiggles rsem-prepare-reference rsem-run-ebseq
+PREFIX ?= /usr/local/bin
+
 
 .PHONY : all ebseq clean
+.PHONY : install
 
 all : $(PROGRAMS)
 
+install : all $(SCRIPTS)
+	install -d "$(PREFIX)/bin"
+	install -m 0755 $(PROGRAMS) $(SCRIPTS) "$(PREFIX)/bin"
+
 sam/libbam.a :
 	cd sam ; ${MAKE} all
 
--- EBSeq/Makefile
+++ EBSeq/Makefile
@@ -1,7 +1,10 @@
 CC = g++
 PROGRAMS = EBSeq rsem-for-ebseq-calculate-clustering-info
+SCRIPTS = rsem-for-ebseq-find-DE rsem-for-ebseq-generate-ngvector-from-clustering-info
+PREFIX ?= /usr/local/bin
 
 .PHONY : all EBSeq clean
+.PHONY : install
 
 all : $(PROGRAMS)
 
@@ -9,7 +12,12 @@ EBSeq :
 	./install
 
 rsem-for-ebseq-calculate-clustering-info : calcClusteringInfo.cpp
-	$(CC) -O3 -Wall calcClusteringInfo.cpp -o $@
+	$(CC) $(CFLAGS) -O3 -Wall calcClusteringInfo.cpp $(LDFLAGS) -o $@
 
 clean : 
 	rm -rf blockmodeling gplots $(PROGRAMS) *~ BiocInstaller
+
+# Assuming EBSeq R package has been installed by other means...
+install : rsem-for-ebseq-calculate-clustering-info $(SCRIPTS)
+	install -d "$(PREFIX)/bin"
+	install -m 0755 $^ "$(PREFIX)/bin"
