--- src/fastqreader.h
+++ src/fastqreader.h
@@ -4,7 +4,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "read.h"
-#include "zlib/zlib.h"
+#include <zlib.h>
 #include "common.h"
 #include <iostream>
 #include <fstream>
--- src/writer.h
+++ src/writer.h
@@ -3,7 +3,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include "zlib/zlib.h"
+#include <zlib.h>
 #include "common.h"
 #include <iostream>
 #include <fstream>
