## Based on modifcations from commit 2be23a4 of upstream repository ##
--- src/delly.cpp
+++ src/delly.cpp
@@ -1939,7 +1939,7 @@
       std::cout << "Usage: " << argv[0] << " [OPTIONS] -g <ref.fa> <sample1.sort.bam> <sample2.sort.bam> ..." << std::endl;
       std::cout << visible_options << "\n"; 
     }
-    return 1; 
+    return 0; 
   }
 
   // Show cmd
--- src/cov.cpp
+++ src/cov.cpp
@@ -281,7 +281,7 @@
       std::cout << "Usage: " << argv[0] << " [OPTIONS] <sample1.bam> <sample2.bam> ..." << std::endl;
       std::cout << visible_options << "\n"; 
     }
-    return 1; 
+    return 0; 
   }
   bool disableRedFilter=false;
   if (vm.count("disable-redundancy")) disableRedFilter=true;
--- src/extract.cpp
+++ src/extract.cpp
@@ -98,7 +98,7 @@
       std::cout << "Usage: " << argv[0] << " [OPTIONS]" << std::endl;
       std::cout << visible_options << "\n"; 
     }
-    return 1; 
+    return 0; 
   }
 
   // Show cmd
--- src/iover.cpp
+++ src/iover.cpp
@@ -419,7 +419,7 @@
       std::cout << "Usage: " << argv[0] << " [OPTIONS] <query_intervals1.txt> <query_intervals2.txt> ..." << std::endl;
       std::cout << visible_options << "\n"; 
     }
-    return 1; 
+    return 0; 
   }
 
   // Resize field identifiers and noIntervals vectors if necessary
--- src/stats.cpp
+++ src/stats.cpp
@@ -613,7 +613,7 @@
       std::cout << "Usage: " << argv[0] << " [OPTIONS] <sample.sort.bam>" << std::endl;
       std::cout << visible_options << "\n"; 
     }
-    return 1; 
+    return 0; 
   }
   if (vm.count("key")) c.text_flag = false;
   else c.text_flag = true;
