--- Makefile
+++ Makefile
@@ -22,19 +22,22 @@
 SYS = UNIX
 
 # Leave blank after "=" to disable; put "= 1" to enable
-WITH_R_PLUGINS = 1
-WITH_WEBCHECK = 1
+WITH_R_PLUGINS =
+WITH_WEBCHECK =
 FORCE_32BIT = 
 WITH_ZLIB = 1
 WITH_LAPACK = 
 FORCE_DYNAMIC = 
 
 # Put C++ compiler here; Windows has it's own specific version
-CXX_UNIX = g++
+CXX_UNIX ?= g++
 CXX_WIN = c:\bin\mingw\bin\mingw32-g++.exe
 
 # Any other compiler flags here ( -Wall, -g, etc)
-CXXFLAGS = 
+CXXFLAGS ?=
+
+# Additional linker flags
+LDFLAGS  ?=
 
 # Misc
 LIB_LAPACK = /usr/lib/liblapack.so.3
@@ -151,7 +154,7 @@
 all : $(OUTPUT) 
 
 $(OUTPUT) :
-	$(CXX) $(CXXFLAGS) -o $(OUTPUT) $(OBJ) $(LIB) 
+	$(CXX) $(CXXFLAGS) -o $(OUTPUT) $(OBJ) $(LDFLAGS) $(LIB) 
 
 $(OBJ) : $(HDR)
 
