--- system.mk/Makefile.darwin
+++ system.mk/Makefile.darwin
@@ -48,7 +48,7 @@
 .SECONDEXPANSION:
 $(SUBDIRS_X86_64): $$(patsubst %_x86_64,${APPDIR}/%,$$@)
 
-all: $(SUBDIRS_I386) $(SUBDIRS_X86_64)
+all: $(SUBDIRS_X86_64)
 
 build: all
 	$(MAKE) build_fat -f $(SELF)
@@ -227,29 +227,29 @@
 # Darwin fat build targets
 #
 sort_c_intel_fat:
-	-test -e ${APPDIR}/sort-bed/bin/sort-bed_i386 && test -e ${APPDIR}/sort-bed/bin/sort-bed_x86_64 && { lipo -create ${APPDIR}/sort-bed/bin/sort-bed_i386 ${APPDIR}/sort-bed/bin/sort-bed_x86_64 -output ${APPDIR}/sort-bed/bin/sort-bed; }
+	cp ${APPDIR}/sort-bed/bin/sort-bed_x86_64 ${APPDIR}/sort-bed/bin/sort-bed
 
 bedops_c_intel_fat:
-	-test -e ${APPDIR}/bedops/bin/bedops_i386 && test -e ${APPDIR}/bedops/bin/bedops_x86_64 && { lipo -create ${APPDIR}/bedops/bin/bedops_i386 ${APPDIR}/bedops/bin/bedops_x86_64 -output ${APPDIR}/bedops/bin/bedops; }
+	cp ${APPDIR}/bedops/bin/bedops_x86_64 ${APPDIR}/bedops/bin/bedops
 
 closestfeatures_c_intel_fat:
-	-test -e ${APPDIR}/closestfeats/bin/closest-features_i386 && test -e ${APPDIR}/closestfeats/bin/closest-features_x86_64 && { lipo -create ${APPDIR}/closestfeats/bin/closest-features_i386 ${APPDIR}/closestfeats/bin/closest-features_x86_64 -output ${APPDIR}/closestfeats/bin/closest-features; }
+	cp ${APPDIR}/closestfeats/bin/closest-features_x86_64 ${APPDIR}/closestfeats/bin/closest-features
 
 bedmap_c_intel_fat:
-	-test -e ${APPDIR}/bedmap/bin/bedmap_i386 && test -e ${APPDIR}/bedmap/bin/bedmap_x86_64 && { lipo -create ${APPDIR}/bedmap/bin/bedmap_i386 ${APPDIR}/bedmap/bin/bedmap_x86_64 -output ${APPDIR}/bedmap/bin/bedmap; }
+	cp ${APPDIR}/bedmap/bin/bedmap_x86_64 ${APPDIR}/bedmap/bin/bedmap
 
 bedextract_c_intel_fat:
-	-test -e ${APPDIR}/bedextract/bin/bedextract_i386 && test -e ${APPDIR}/bedextract/bin/bedextract_x86_64 && { lipo -create ${APPDIR}/bedextract/bin/bedextract_i386 ${APPDIR}/bedextract/bin/bedextract_x86_64 -output ${APPDIR}/bedextract/bin/bedextract; }
+	cp ${APPDIR}/bedextract/bin/bedextract_x86_64 ${APPDIR}/bedextract/bin/bedextract
 
 convert2bed_c_intel_fat:
-	-test -e ${APPDIR}/conversion/bin/convert2bed_i386 && test -e ${APPDIR}/conversion/bin/convert2bed_x86_64 && { lipo -create ${APPDIR}/conversion/bin/convert2bed_i386 ${APPDIR}/conversion/bin/convert2bed_x86_64 -output ${APPDIR}/conversion/bin/convert2bed; }
+	cp ${APPDIR}/conversion/bin/convert2bed_x86_64 ${APPDIR}/conversion/bin/convert2bed
 
 starch_c_intel_fat:
 	cp ${APPDIR}/starch/src/starchcluster_sge.tcsh ${APPDIR}/starch/bin/starchcluster_sge
 	cp ${APPDIR}/starch/src/starchcluster_gnuParallel.tcsh ${APPDIR}/starch/bin/starchcluster_gnuParallel
-	-test -e ${APPDIR}/starch/bin_i386/starch && test -e ${APPDIR}/starch/bin_x86_64/starch && { lipo -create ${APPDIR}/starch/bin_i386/starch ${APPDIR}/starch/bin_x86_64/starch -output ${APPDIR}/starch/bin/starch; }
-	-test -e ${APPDIR}/starch/bin_i386/unstarch && test -e ${APPDIR}/starch/bin_x86_64/unstarch && { lipo -create ${APPDIR}/starch/bin_i386/unstarch ${APPDIR}/starch/bin_x86_64/unstarch -output ${APPDIR}/starch/bin/unstarch; }
-	-test -e ${APPDIR}/starch/bin_i386/starchcat && test -e ${APPDIR}/starch/bin_x86_64/starchcat && { lipo -create ${APPDIR}/starch/bin_i386/starchcat ${APPDIR}/starch/bin_x86_64/starchcat -output ${APPDIR}/starch/bin/starchcat; }
+	cp ${APPDIR}/starch/bin_x86_64/starch ${APPDIR}/starch/bin/starch
+	cp ${APPDIR}/starch/bin_x86_64/unstarch ${APPDIR}/starch/bin/unstarch
+	cp ${APPDIR}/starch/bin_x86_64/starchcat ${APPDIR}/starch/bin/starchcat
 
 
 
@@ -257,26 +257,26 @@
 # Debug: Darwin fat build targets
 #
 dbg_sort_c_intel_fat:
-	-test -e ${APPDIR}/sort-bed/bin/debug.sort-bed_i386 && test -e ${APPDIR}/sort-bed/bin/debug.sort-bed_x86_64 && { lipo -create ${APPDIR}/sort-bed/bin/debug.sort-bed_i386 ${APPDIR}/sort-bed/bin/debug.sort-bed_x86_64 -output ${APPDIR}/sort-bed/bin/debug.sort-bed; }
+	cp ${APPDIR}/sort-bed/bin/debug.sort-bed_x86_64 ${APPDIR}/sort-bed/bin/debug.sort-bed
 
 dbg_bedops_c_intel_fat:
-	-test -e ${APPDIR}/bedops/bin/debug.bedops_i386 && test -e ${APPDIR}/bedops/bin/debug.bedops_x86_64 && { lipo -create ${APPDIR}/bedops/bin/debug.bedops_i386 ${APPDIR}/bedops/bin/debug.bedops_x86_64 -output ${APPDIR}/bedops/bin/debug.bedops; }
+	cp ${APPDIR}/bedops/bin/debug.bedops_x86_64 ${APPDIR}/bedops/bin/debug.bedops
 
 dbg_closestfeatures_c_intel_fat:
-	-test -e ${APPDIR}/closestfeats/bin/debug.closest-features_i386 && test -e ${APPDIR}/closestfeats/bin/debug.closest-features_x86_64 && { lipo -create ${APPDIR}/closestfeats/bin/debug.closest-features_i386 ${APPDIR}/closestfeats/bin/debug.closest-features_x86_64 -output ${APPDIR}/closestfeats/bin/debug.closest-features; }
+	cp ${APPDIR}/closestfeats/bin/debug.closest-features_x86_64 ${APPDIR}/closestfeats/bin/debug.closest-features
 
 dbg_bedmap_c_intel_fat:
-	-test -e ${APPDIR}/bedmap/bin/debug.bedmap_i386 && test -e ${APPDIR}/bedmap/bin/debug.bedmap_x86_64 && { lipo -create ${APPDIR}/bedmap/bin/debug.bedmap_i386 ${APPDIR}/bedmap/bin/debug.bedmap_x86_64 -output ${APPDIR}/bedmap/bin/debug.bedmap; }
+	cp ${APPDIR}/bedmap/bin/debug.bedmap_x86_64 ${APPDIR}/bedmap/bin/debug.bedmap
 
 dbg_bedextract_c_intel_fat:
-	-test -e ${APPDIR}/bedextract/bin/debug.bedextract_i386 && test -e ${APPDIR}/bedextract/bin/debug.bedextract_x86_64 && { lipo -create ${APPDIR}/bedextract/bin/debug.bedextract_i386 ${APPDIR}/bedextract/bin/debug.bedextract_x86_64 -output ${APPDIR}/bedextract/bin/debug.bedextract; }
+	cp ${APPDIR}/bedextract/bin/debug.bedextract_x86_64 ${APPDIR}/bedextract/bin/debug.bedextract
 
 dbg_convert2bed_c_intel_fat:
-	-test -e ${APPDIR}/conversion/bin/debug.convert2bed_i386 && test -e ${APPDIR}/conversion/bin/debug.convert2bed_x86_64 && { lipo -create ${APPDIR}/conversion/bin/debug.convert2bed_i386 ${APPDIR}/conversion/bin/debug.convert2bed_x86_64 -output ${APPDIR}/conversion/bin/debug.convert2bed; }
+	cp ${APPDIR}/conversion/bin/debug.convert2bed_x86_64 ${APPDIR}/conversion/bin/debug.convert2bed
 
 dbg_starch_c_intel_fat:
 	cp ${APPDIR}/starch/src/starchcluster_sge.tcsh ${APPDIR}/starch/bin/starchcluster_sge
 	cp ${APPDIR}/starch/src/starchcluster_gnuParallel.tcsh ${APPDIR}/starch/bin/starchcluster_gnuParallel
-	-test -e ${APPDIR}/starch/bin_i386/debug.starch && test -e ${APPDIR}/starch/bin_x86_64/debug.starch && { lipo -create ${APPDIR}/starch/bin_i386/debug.starch ${APPDIR}/starch/bin_x86_64/debug.starch -output ${APPDIR}/starch/bin/debug.starch; }
-	-test -e ${APPDIR}/starch/bin_i386/debug.unstarch && test -e ${APPDIR}/starch/bin_x86_64/debug.unstarch && { lipo -create ${APPDIR}/starch/bin_i386/debug.unstarch ${APPDIR}/starch/bin_x86_64/debug.unstarch -output ${APPDIR}/starch/bin/debug.unstarch; }
-	-test -e ${APPDIR}/starch/bin_i386/debug.starchcat && test -e ${APPDIR}/starch/bin_x86_64/debug.starchcat && { lipo -create ${APPDIR}/starch/bin_i386/debug.starchcat ${APPDIR}/starch/bin_x86_64/debug.starchcat -output ${APPDIR}/starch/bin/debug.starchcat; }
+	cp ${APPDIR}/starch/bin_x86_64/debug.starch ${APPDIR}/starch/bin/debug.starch
+	cp ${APPDIR}/starch/bin_x86_64/debug.unstarch ${APPDIR}/starch/bin/debug.unstarch
+	cp ${APPDIR}/starch/bin_x86_64/debug.starchcat ${APPDIR}/starch/bin/debug.starchcat
