--- hmmer/src/impl_sse/p7_oprofile.c
+++ hmmer/src/impl_sse/p7_oprofile.c
@@ -670,7 +670,7 @@
   uint8_t b;
 
   sc  = -1.0f * roundf(om->scale_b * sc);                          /* ugh. sc is now an integer cost represented in a float...           */
-  b   = (sc > 255 - om->bias_b) ? 255 : (uint8_t) sc + om->bias_b; /* and now we cast, saturate, and bias it to an unsigned char cost... */
+  b   = (sc > 255 - om->bias_b) ? 255 : (uint8_t)(signed int) sc + om->bias_b; /* and now we cast, saturate, and bias it to an unsigned char cost... */
   return b;
 }
  
@@ -686,7 +686,7 @@
   uint8_t b;
 
   sc  = -1.0f * roundf(om->scale_b * sc);       /* ugh. sc is now an integer cost represented in a float...    */
-  b   = (sc > 255.) ? 255 : (uint8_t) sc;	/* and now we cast and saturate it to an unsigned char cost... */
+  b   = (sc > 255.) ? 255 : (uint8_t)(signed int) sc;	/* and now we cast and saturate it to an unsigned char cost... */
   return b;
 }
  
